modules/misunderstood/src/backend/db.ts

```diff
@@ -178,4 +170,18 @@ export default class Db {
   applyChanges(botId: string) {
     return applyChanges(this.bp, botId, TABLE_NAME)
   }
+
+  filterQuery(query, options?: { startDate: Date; endDate: Date; reason?: string }) {
```

Comment:
one last comment, you're always declaring inline interface for options can you declare an explicit interface in typings and use it everywhere instead

```ts
type FilteringOptions = Partial<{
 startDate: Date
 endDate: Date
 reason: string
}>
```

Reply:
done